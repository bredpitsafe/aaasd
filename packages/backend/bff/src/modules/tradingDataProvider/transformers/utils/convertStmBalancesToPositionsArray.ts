import type { TStmBalance } from '@grpc-schemas/trading_data_provider-api-sdk/services/trading_data_provider/v1/stm_api.js';
import { omit } from 'lodash-es';

import type { TStmPosition } from '../../schemas/SubscribeToStmPositions.schema.ts';

export const convertStmBalancesToPositionsArray = (balances?: TStmBalance[]): TStmPosition[] =>
    balances?.map((update) => omit(update, 'balances')) ?? [];
