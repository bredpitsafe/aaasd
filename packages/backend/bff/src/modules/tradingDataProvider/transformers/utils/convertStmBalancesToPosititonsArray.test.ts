import { describe, expect, it } from 'vitest';

import { stmBalance } from '../fixtures/stmBalance.ts';
import { convertStmBalancesToPositionsArray } from './convertStmBalancesToPositionsArray.ts';

describe('convertStmBalancesToPositionsArray', () => {
    it('converts data correctly', () => {
        expect(convertStmBalancesToPositionsArray([stmBalance])).toMatchSnapshot();
    });
    it('converts undefined to empty array', () => {
        expect(convertStmBalancesToPositionsArray()).toEqual([]);
    });
});
