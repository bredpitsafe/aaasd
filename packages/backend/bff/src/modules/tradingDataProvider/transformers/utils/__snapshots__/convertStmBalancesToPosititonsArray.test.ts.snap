// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`convertStmBalancesToPositionsArray > converts data correctly 1`] = `
[
  {
    "instrumentId": 456,
    "instrumentName": "INSTRUMENT_ID",
    "platformTime": "2024-02-26 01:59:47.376",
    "position": 123,
    "robotId": 789,
    "robotName": "ROBOT_NAME",
    "virtualAccountId": 123,
    "virtualAccountName": "VIRTUAL_ACCOUNT",
  },
]
`;
